### ISSUE TITLE
Provide working example of remote source in documentation

### ISSUE THREAD

**Issue by c256985** [ellipticaljs/paper-autocomplete#97]
I can't seem to figure out how the remote source functionality is supposed to work. My thought is that if you supply a json file or REST endpoint, along with a text and property value, then it should extract these attributes from the response, and add them as suggestions to the autocomplete list.

```
<paper-autocomplete label="Users" autofocus remote-source source="/data/userlist.json" text-property="name" value-property="id">
```

Where the json file looks something like this:
```
[ {"id": 1, "name":"John Smith"}, {"id":2, "name":"Jane Doe"}]

```

But this doesn't seem to work. Am I missing something?

**Comment by srijken** [comment]
👍

**Comment by leifneve** [comment]
The source property is ignored if remote-source is used. The documentation says to listen to changes using the change event and provide suggestions using the suggestions() method. That is working for me.

**Comment by MrMcGibblets** [comment]
Considering I just found @leifneve comment extremely useful as I was confused with the documentation, I would recommend either an example or just a little paragraph.

**Comment by jhuesos** [comment]
Agree, I am not very happy with the remote-source implementation, I think it is kind of confusing. I will try to add the example in the documentation.

Whenever I have time, my plan is to re-write the component to be only Polymer 2 compatible, to make it faster. My idea is to simplify it and change the way remote-source works. One of the ideas I have is to remove the remote-source option and just change queryFn to accept functions that returns promises. When queryFn returns promises it is assumed that it is a remote-search and the component will work just fine. 

But once I find the time to start with new implementation, I will open an issue to collect feedback.

**Comment by ndormont** [comment]
queryFn + Promise is definitely the way to go! great idea.
